Source/WebCore/workers/WorkerConsoleClient.cpp

```diff
@@ -117,11 +143,86 @@ void WorkerConsoleClient::timeStamp(JSC::JSGlobalObject*, Ref<ScriptArguments>&&
 
-// FIXME: <https://webkit.org/b/243361> Web Inspector: support console screenshots in a Worker
-void WorkerConsoleClient::screenshot(JSC::JSGlobalObject*, Ref<ScriptArguments>&&) { }
+void WorkerConsoleClient::screenshot(JSC::JSGlobalObject* lexicalGlobalObject, Ref<ScriptArguments>&& arguments)
+{
+    // FIXME: <https://webkit.org/b/217724> Add support for WorkletGlobalScope.
+    if (!is<WorkerGlobalScope>(m_globalScope))
+        return;
+
+    JSC::VM& vm = lexicalGlobalObject->vm();
+    String dataURL;
+    JSC::JSValue target;
+
+    auto timestamp = WallTime::now();
+
+    if (arguments->argumentCount()) {
+        auto possibleTarget = arguments->argumentAt(0);
+
+        if (auto* imageData = JSImageData::toWrapped(vm, possibleTarget)) {
+            target = possibleTarget;
+            if (InspectorInstrumentation::hasFrontends()) [[unlikely]] {
+                if (auto imageBuffer = ImageBuffer::create(imageData->size(), RenderingMode::Unaccelerated, RenderingPurpose::Unspecified, /* scale */ 1, DestinationColorSpace::SRGB(), PixelFormat::BGRA8)) {
+                    imageBuffer->putPixelBuffer(imageData->byteArrayPixelBuffer().get(), IntRect(IntPoint(), imageData->size()));
+                    dataURL = imageBuffer->toDataURL("image/png"_s, /* quality */ std::nullopt, PreserveResolution::Yes);
+                }
+            }
+        } else if (auto* imageBitmap = JSImageBitmap::toWrapped(vm, possibleTarget)) {
+            target = possibleTarget;
+            if (InspectorInstrumentation::hasFrontends()) [[unlikely]] {
+                if (auto* imageBuffer = imageBitmap->buffer())
+                    dataURL = imageBuffer->toDataURL("image/png"_s, /* quality */ std::nullopt, PreserveResolution::Yes);
+            }
+        } else if (auto* context = canvasRenderingContext(vm, possibleTarget)) {
+            target = possibleTarget;
+            if (InspectorInstrumentation::hasFrontends()) [[unlikely]] {
+                if (auto result = InspectorCanvas::getContentAsDataURL(*context))
+                    dataURL = result.value();
+            }
+        } else {
+            String base64;
+            if (possibleTarget.getString(lexicalGlobalObject, base64) && base64.length() > 5 && startsWithLettersIgnoringASCIICase(base64, "data:"_s)) {
+                target = possibleTarget;
+                dataURL = base64;
+            }
+        }
+    }
+
+    if (InspectorInstrumentation::hasFrontends()) [[unlikely]] {
+        if (dataURL.isEmpty()) {
+            InspectorInstrumentation::addMessageToConsole(m_globalScope, makeUnique<Inspector::ConsoleMessage>(MessageSource::ConsoleAPI, MessageType::Image, MessageLevel::Error, "Could not capture screenshot"_s, WTFMove(arguments)));
+            return;
+        }
+    }
+
+    Vector<JSC::Strong<JSC::Unknown>> adjustedArguments;
+    adjustedArguments.reserveInitialCapacity(arguments->argumentCount());
```

Comment:
This allocates initial capacity that is 1 too small when `target` is `nullptr`, resulting in a vector that is too big when the append overflows the initial capacity. Here is one at to correct that, but maybe you can find a better way.
```suggestion
    adjustedArguments.reserveInitialCapacity(arguments->argumentCount() + !target);
```

Maybe we can share code so this tricky thing doesn't have to be written out twice?